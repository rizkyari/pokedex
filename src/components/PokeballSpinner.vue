<template>
    <div>
        <div class="body-wrapper">
            <div class="pokemon"></div>
        </div>
    </div>
    <div class="my-3">
        <h3>Please Wait...</h3>
    </div>
</template>

<script setup></script>

<style scoped>
.body-wrapper {
    margin: 0;
    display: flex;
    justify-content: center;
    align-items: center;
}
.pokemon {
  position: relative;
  height: 100px;
  width: 100px;
  background: linear-gradient(to bottom, rgb(254, 0, 1) 50%, white 50% );
  border-radius: 50%;
  border: 8px solid black;
  animation: spin 1s linear infinite;
}

.pokemon::before {
  content: '';
  position: absolute;
  height: 8px;
  width: 100%;
  background: black;
  top: 50%;
  left: 0;
  transform: translatey(-50%);
}

.pokemon::after {
  content: '';
  position: absolute;
  height: 38px;
  width: 38px;
  border-radius: 50%;
  background: white;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  box-shadow: inset 0 0 0 8px black, inset 0 0 0 10px white, inset 0 0 0 12px black;

}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
</style>