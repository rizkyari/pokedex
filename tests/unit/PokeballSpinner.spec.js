import { mount } from '@vue/test-utils';
import PokeballSpinner from '@/components/PokeballSpinner.vue';

describe('PokeballSpinner', () => {
  it('renders the component correctly', () => {
    const wrapper = mount(PokeballSpinner);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('.pokemon').exists()).toBe(true);
    expect(wrapper.find('h3').text()).toBe('Please Wait...'); 
  });

  it('matches snapshot', () => {
    const wrapper = mount(PokeballSpinner);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
