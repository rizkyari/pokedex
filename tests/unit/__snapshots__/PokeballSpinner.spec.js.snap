// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeballSpinner matches snapshot 1`] = `
<div>
  <div class="body-wrapper">
    <div class="pokemon"></div>
  </div>
</div>
<div class="my-3">
  <h3>Please Wait...</h3>
</div>
`;
